### ISSUE TITLE
Feature: add AI logic

### ISSUE THREAD

**Issue by David52920** [plaza-in-a-heatwave/Cadesim#78]
Have several different types of AI:

1. Sit & shoot type (near flag clusters) 

2. Battle type (AI that reads player moves at beginning of turn nearest to them and try to sink them)

- Possible different battle types based on difficulty (more aggressive or conservative; use less moves and shots)

3. Flag collecting type (check position on board where cluster has more points)

- Shoot players if near
---------------------------------------
Allow players to specify #of bots and difficulty.

**Comment by mdjong1** [comment]
Happen to have an idea of basic logic steps for each AI?

**Comment by David52920** [comment]
> Happen to have an idea of basic logic steps for each AI?
> 
> 
> 
> "once upon a time" I set up this as idea for what I thought it should be. Tbh would probably change with current knowledge
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/4410453/95958041-d6b66a80-0e00-11eb-9035-8b249c1b2d1d.png)
> 
> (This stems from 26th Jan 2018)

No, haven't really had time to look into AI yet. Been working on other things like map editor and some collision bugs and such.
